Lib/tarfile.py

```diff
@@ -207,8 +207,9 @@ def itn(n, digits=8, format=DEFAULT_FORMAT):
             s = bytearray([0o200])
         else:
             s = bytearray([0o377])
-            n = 256 ** digits + n
+            n = 256 ** digits + int(n)
 
+        n = int(n)
```

Comment:
Wouldn't be better to convert `n` to `int` at the begin of the function?

Reply:
My concern in doing so was that the loss of precision might lead to unwanted results if it is done before the comparisons (e.g. -0.1 becomes 0). That said, it may not matter in any practical case.

Reply:
This is a good example why this should be done before the comparisons. With this patch itn() produces incorrect results for values between -1 and 0.
```
>>> tarfile.itn(-0.1)
bytearray(b'\xff\x00\x00\x00\x00\x00\x00\x00')
>>> tarfile.nti(tarfile.itn(-0.1))
-72057594037927936
```
Please add this case to tests.